libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -808,11 +816,13 @@ proc maintainDirectPeer(g: GossipSub, id: PeerId, addrs: seq[MultiAddress]) {.as
     except CatchableError as exc:
       debug "Direct peer error dialing", description = exc.msg
 
-proc addDirectPeer*(g: GossipSub, id: PeerId, addrs: seq[MultiAddress]) {.async.} =
+proc addDirectPeer*(
+    g: GossipSub, id: PeerId, addrs: seq[MultiAddress]
+) {.async: (raises: [CancelledError]).} =
   g.parameters.directPeers[id] = addrs
   await g.maintainDirectPeer(id, addrs)
 
-proc maintainDirectPeers(g: GossipSub) {.async.} =
+proc maintainDirectPeers(g: GossipSub) {.async: (raises: [CancelledError]).} =
```

Comment:
often when introducing `raises`, it also makes sense to make a pass over callers to functions to ensure that they no longer contain unnecessary try/except - not sure if that's the case, but it's worth investigating. In particular, we strive to avoid `except CatchableError` in code except where it deliberately is the right thing to do ("catch exceptions from any abstraction level").